### PR TITLE
feat(query-generator): Generate UPDATEs using bind parameters

### PR DESCRIPTION
**Pull request by gazoakley**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

WIP changes to allow UPDATEs to use bind parameters instead of directly included values. For example, instead of:

```UPDATE `myTable` SET `name`='foo',`birthday`='2011-03-27 10:01:55' WHERE `id` = 2```

Generate:

```UPDATE `myTable` SET `name`=$1,`birthday`=$2 WHERE `id` = $3```

(with parameters `'foo'`, `'2011-03-27 10:01:55'` and `2`)

### CODE REVIEW

File: test/integration/data-types.test.js [hunk 1]
```diff
@@ -175,9 +175,12 @@ describe(Support.getTestDialectTeaser('DataTypes'), () => {
     return testSuccess(Type, 'foobar');
   });
 
-  it('calls parse and stringify for STRING', () => {
+  it('calls parse and stringify/bindParam for STRING', () => {
     const Type = new Sequelize.STRING();
 
+    if (dialect === 'mssql') {
```
Comment (sushantdhiman) [inline test/integration/data-types.test.js]: Why only mssql?

File: test/integration/model/create.test.js [hunk 2]
```diff
@@ -984,8 +984,8 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           return user.save({
             logging(sql) {
               test = true;
-              expect(sql).to.contain('ARRAY[]::INTEGER[]');
-              expect(sql).to.contain('ARRAY[]::VARCHAR(255)[]');
+              expect(sql).not.to.contain('ARRAY[]::INTEGER[]');
```
Comment (sushantdhiman) [inline test/integration/model/create.test.js]: These tests should be moved to unit tests, but that is too much work, should be tackled in different PR. Please add some assertion which covers what is expected here, like may be just `ARRAY[]`?

File: test/integration/data-types.test.js (hunk 1 above)
Comment (gazoakley) [reply test/integration/data-types.test.js]: mssql has a `_bindParam` function that outputs either a Buffer or a normal string dependent on whether or not the `STRING` data type uses the binary param. This override isn't needed for other dialects - I've added a comment.

File: test/integration/model/create.test.js (hunk 2 above)
Comment (gazoakley) [reply test/integration/model/create.test.js]: I've tidied this up a bit:

* https://github.com/sequelize/sequelize/pull/9492/files#diff-4b224bbc09dfb867f8671d77e298bb8cR955
* https://github.com/sequelize/sequelize/pull/9492/files#diff-4b224bbc09dfb867f8671d77e298bb8cR955

**Comment by gazoakley** [comment]
@sushantdhiman @janmeier Thanks 🎉

**Comment by sushantdhiman** [comment]
Thanks for this contribution @gazoakley , This is an important missing feature from Sequelize
